src/python/TaskWorker/Actions/Splitter.py

```diff
@@ -22,28 +22,36 @@ def execute(self, *args, **kwargs):
 
         maxJobs = getattr(self.config.TaskWorker, 'maxJobsPerTask', 10000)
 
+        data = args[0]
         splitparam = kwargs['task']['tm_split_args']
         splitparam['algorithm'] = kwargs['task']['tm_split_algo']
         if kwargs['task']['tm_job_type'] == 'Analysis':
+            totalUnits = kwargs['task']['tm_totalunits']
             if kwargs['task']['tm_split_algo'] == 'FileBased':
-                splitparam['total_files'] = kwargs['task']['tm_totalunits']
+                if totalUnits < 1.0:
+                    totalUnits *= len(data.getFiles())
```

Comment:
do we need to round to an integer ?

Reply:
also in the following lines

Reply:
I think tentatively not, [as WMCore seems to do this](https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/JobSplitting/FileBased.py#L28).

Reply:
well.. unless that int in WMCore is part of their public interace i.e. a clear commitment, I'd rather not rely on it and for mind sanity round to integer here.

Reply:
Ok